PowerUp/app/src/main/java/powerup/systers/com/AvatarActivity.java

```diff
@@ -74,17 +75,36 @@ public void onCreate(Bundle savedInstanceState) {
 		continueButton.setOnClickListener(new View.OnClickListener() {
 			@Override
 			public void onClick(View v) {
-                SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(AvatarActivity.this);
-                boolean hasPreviouslyStarted = prefs.getBoolean(getString(R.string.preferences_has_previously_started), false);
-                if (!hasPreviouslyStarted) {
-                    SharedPreferences.Editor edit = prefs.edit();
-                    edit.putBoolean(getString(R.string.preferences_has_previously_started), Boolean.TRUE);
-                    edit.apply();
-                }
-                Intent myIntent = new Intent(AvatarActivity.this, MapActivity.class);
-				AvatarRoomActivity.avatarRoomInstance.finish();
+				if(fromActivity == 1) {
+					SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(AvatarActivity.this);
+					boolean hasPreviouslyStarted = prefs.getBoolean(getString(R.string.preferences_has_previously_started), false);
+					if (!hasPreviouslyStarted) {
+						SharedPreferences.Editor edit = prefs.edit();
+						edit.putBoolean(getString(R.string.preferences_has_previously_started), Boolean.TRUE);
+						edit.apply();
+					}
+					Intent myIntent = new Intent(AvatarActivity.this, MapActivity.class);
+					AvatarRoomActivity.avatarRoomInstance.finish();
+					finish();
+                   // myIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
+					startActivityForResult(myIntent, 0);
+				}
+				else{
+					Intent myIntent = new Intent(AvatarActivity.this, MapActivity.class);
```

Comment:
Give intent descriptive name and also it can be dynamically specified at call time in startActivity(new Intent()) way.